File: HR/ClientApp/src/app/pages/company/shared/company.model.ts
import { IEntity } from '../../shared/entity.model';
import { IAuditable } from '../../shared/auditable.model';

export interface ICompany extends IEntity, IAuditable {
    companyName:string;
    address :string;
    phone :string;
    fax :string;
    email :string;
    contactPerson :string;
    contactPersonEmail :string;
    contactPersonPhone :string;
    fiscalYearStart :string;
}

export class ISaveCompany{
    id: string;
    companyName:string;
    address :string;
    phone :string;
    fax :string;
    email :string;
    contactPerson :string;
    contactPersonEmail :string;
    contactPersonPhone :string;
    fiscalYearStart :string;
}

File: HR/Models/CompanyModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HR.Models
{
    public class CompanyModel
    {
        public Guid Id { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public Guid? CreatedById { get; set; }
        public Guid? UpdatedById { get; set; }
        public string CompanyName{ get;set;}
        public string Address { get; set; }
        public string Phone { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string ContactPerson { get; set; }
        public string ContactPersonEmail { get; set; }
        public string ContactPersonPhone { get; set; }
        public string FiscalYearStart { get; set; }
    
    }
     public class CompanySaveModel
    {
        public Guid Id { get; set; }
        public string CompanyName{ get;set;}
        public string Address { get; set; }
        public string Phone { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string ContactPerson { get; set; }
        public string ContactPersonEmail { get; set; }
        public string ContactPersonPhone { get; set; }
        public string FiscalYearStart { get; set; }
    }
}